### PR TITLE
docs: replace dead magic bitboard link with wikipedia

### PR DESCRIPTION
**Pull request by kunal-mandalia**
Great work in Lichess. Was looking through the code out of interest and noticed a dead link so here's a fix.

---
This change is [Reviewable](https://reviewable.io/reviews/lichess-org/compression/7)

### CODE REVIEW

**Comment by ddugovic** [comment]
Could we perhaps use https://www.chessprogramming.org/Magic_Bitboards ?

**Comment by kunal-mandalia** [comment]
@ddugovic Sure - that's much more thorough

File: src/main/java/game/Magic.java [hunk 1]
```diff
@@ -17,7 +17,7 @@ public Magic(long mask, long factor, int offset) {
     }
 
     // Precomputed overlapping fixed shift magics:
-    // https://en.wikipedia.org/wiki/Bitboard#Magic_bitboards
+    //  https://www.chessprogramming.org/Magic_Bitboards
```
Comment (niklasf) [inline src/main/java/game/Magic.java]: Could you please change to https://www.chessprogramming.org/Magic_Bitboards#Fixed_shift_Fancy (that's the specific appraoch that's used here) and remove the double leading space?

**Comment by niklasf** [comment]
Thanks!

**Pull request merged**